fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java

```diff
@@ -1384,4 +1386,242 @@ private SqlQueryData processMissingParm(Class<?> resourceType, QueryParameter qu
+                        // Build this join: JOIN <modifierTypeResourceName>_<type>_VALUES AS CPx
+                        //                    ON CPx.LOGICAL_RESOURCE_ID = CLR<x-1>.LOGICAL_RESOURCE_ID
+                        //                    AND
+                        whereClauseSegment.append(JOIN + QuerySegmentAggregator.tableName(previousParm.getModifierResourceTypeName(), currentParm))
+                                            .append(" AS " + chainedParmVar + ON + chainedParmVar + DOT + "LOGICAL_RESOURCE_ID");
+                    } else {
+                        // Build this join: JOIN <modifierTypeResourceName>_LOGICAL_RESOURCES AS CLRx
+                        //                    ON CLRx.LOGICAL_RESOURCE_ID = CLR<x-1>.LOGICAL_RESOURCE_ID
+                        //                    AND
+                        whereClauseSegment.append(JOIN + previousParm.getModifierResourceTypeName() + "_LOGICAL_RESOURCES")
+                                            .append(" AS " + chainedLogicalResourceVar + ON + chainedLogicalResourceVar + DOT + "LOGICAL_RESOURCE_ID");
+                    }
+                    whereClauseSegment.append(EQ + prevChainedLogicalResourceVar + DOT + "LOGICAL_RESOURCE_ID" + AND);
+                }
+
+                // Build the rest
+                SqlQueryData sqlQueryData;
+                if (!"_id".equals(currentParm.getCode())) {
+                    // (CPx.PARAMETER_NAME_ID=<code-id> AND (CPx.<type>_VALUE=<valueCode>))
+                    sqlQueryData = buildQueryParm(ModelSupport.getResourceType(previousParm.getModifierResourceTypeName()), currentParm, chainedParmVar);
+                } else {
+                    // CLRx.LOGICAL_ID IN (?)
+                    sqlQueryData = buildChainedIdClause(currentParm, chainedParmVar);
+                }
+
+                if (log.isLoggable(Level.FINE)) {
+                    log.fine("chained sqlQueryData[" + chainedParmVar + "] = " + sqlQueryData.getQueryString());
+                }
+                whereClauseSegment.append(sqlQueryData.getQueryString());
+                bindVariables.addAll(sqlQueryData.getBindVariables());
+           }
+
+            // Insert where clause segment in whole
+            whereClauseSegments.insert(0, whereClauseSegment.toString());
+
+            previousParm = currentParm;
+            parmIndex++;
+        }
+
+        SqlQueryData queryData = new SqlQueryData(whereClauseSegments.insert(0, selectSegments.toString()).toString(), bindVariables);
+        log.exiting(CLASSNAME, METHODNAME, queryData.getQueryString());
+        return queryData;
+    }
+
+    /**
+     * Populates the reference parameter name ID and code system ID sub-segment of the passed where clause segment.
+     *
+     * @param whereClauseSegment - the segment to which the sub-segment will be added
+     * @param queryParmName - the search parameter name
+     * @param resourceTypeName - the resource type of the reference being followed
```

Comment:
Not to change too much, but please note we use code not names. I think for consistency, we probably leave as name here.

Reply:
noted.